### DISCUSSION TITLE
Controlling Faikin via MQTT

### DISCUSSION THREAD

**Discussion by staub79** [revk/xESP32-Faikout#361] [General]
Hello,
I got the Faikin code working on a ESP32 and I have to say it works like a charme.
Only with the MQTT commanding Im struggeling a bit.
Currently I can set the power (on/off) by sending 0 or 1 to Faikin/[Hostname]/command/power
However some other aircon controls will not change via mqtt.
For example EconoMode by sending true/false to Faikin/[Hostname]/command/econo
Does anyone have an idea how econo mode can be changed via MQTT?
I have similar problems with swingh/swingv/streamer/powerfull/quietoutdoor/comfort/led/sensormode
Greetings
Michael Bartsch

**Comment by revk** [comment]
The main command is `control` which takes JSON for the settings. This is what is mainly intended for MQTT settings, so sending `control` with `{"powerful":true}`, etc.

There are also a number of simple direct commands with no settings: `on`, `off`, `auto`, `heat`, `cool`, `dry`, `fan`, `low`, `medium`, `high`.

Then there are a load of commands with arguments intended for Home Assistant usage, and updated as we expand HA config: `temp`, `mode`, `fan`, `swing`, `preset`, `demand`, `sensor`, `power`, `streamer` though the last few are only in latest beta release.

**Reply by staub79** [reply]
Thanks for the quick answer and the good hint.
Im now sending the {"powerful":true} to Faikin/[Hostname]/command/control and now it works.
At first I thought it would be Faikin/[Hostname]/control which did not work since the command was missing.
Thanks for your help and especially the Faikin source code.
With this upcoming API change and the limit of 200 changes per day my automation would not work anymore.
You made my day :)

BTW you can add the unit FTXM25R2V1B to the list of working aircon's
I just had to use the free S21 port and a PAP connector.

**Reply by revk** [reply]
You should be able to edit the wiki.

The exact structure of the MQTT topic is subject to some settings, sorry, so could not say exactly what it would be in your case.
